tests/ConformanceTest-report.txt

```diff
@@ -1,11 +1,11 @@
-# 5 pass; 14 fail; 19 total; 26.3% score
-FAIL: Basic.java:28 test:expression-type:Object?:nullable
-FAIL: Basic.java:28 test:sink-type:Object!:return
+# 11 pass; 8 fail; 19 total; 57.9% score
+PASS: Basic.java:28 test:expression-type:Object?:nullable
+PASS: Basic.java:28 test:sink-type:Object!:return
 PASS: Basic.java:28 test:cannot-convert:Object? to Object!
-FAIL: Basic.java:34 test:expression-type:Object!:nonNull
-FAIL: Basic.java:34 test:sink-type:Object?:return
-FAIL: Basic.java:41 test:sink-type:Object?:nullableObject
-FAIL: Basic.java:43 test:sink-type:String!:testSinkType#nonNullString
+PASS: Basic.java:34 test:expression-type:Object!:nonNull
+PASS: Basic.java:34 test:sink-type:Object?:return
+PASS: Basic.java:41 test:sink-type:Object?:nullableObject
+PASS: Basic.java:43 test:sink-type:String!:testSinkType#nonNullString
 FAIL: Basic.java:49 test:expression-type:List!<capture of ? extends String?>:nullableStrings
 FAIL: Basic.java: no unexpected facts
```

Comment:
This is the last mis-match against `main`. I manually looked into the extra error and I think it is:

````
../jspecify/conformance-tests/src/assertions/java/org/jspecify/conformance/tests/Basic.java:47: error: (type.argument.type.incompatible) $$ 4 $$ E $$ List $$ capture#85 of ? extends String* $$ Object* $$ ( 1525, 1551 ) $$ incompatible type argument for type parameter E of List.
  Object testWildcard(List<? extends @Nullable String> nullableStrings) {
                           ^
  found   : capture#85 of ? extends String*
  required: Object*
````

Here, `Basic.java` failed already before, so that's fine.

However, for `tests/ConformanceTestOnSamples-report.txt` there are now many more failing tests because of `FAIL: AnnotatedReceiver.java: no unexpected facts`, which previously passed.
How can the changes in this PR cause more unexpected facts? It's only outputting type information as warnings, which should be ignored.

@netdpb Do you understand why this happens?
How can I output the unexpected facts most conveniently?

Reply:
That would be any extra errors.

You can see them by [setting the `JSPECIFY_CONFORMANCE_TEST_MODE` environment variable](https://github.com/jspecify/jspecify-reference-checker/blob/ec180c5af659ac9c242672a8a0e84c63090dd195/conformance-test-framework/src/main/java/org/jspecify/conformance/ConformanceTestRunner.java#L157) to "details".

Reply:
When I try that, I see lots of warnings about `@NullnessUnspecified`:

````
    OOPS: NotNullMarkedAnnotatedWildcardUnspec.java:17 () cannot find symbol
    import org.jspecify.annotations.NullnessUnspecified;
                                   ^
      symbol:   class NullnessUnspecified
      location: package org.jspecify.annotations
````

Do you see those on `main`?

Also, there are many messages like this:

````
    Jan. 16, 2024 6:12:47 P.M. tests.ConformanceTest$DetailMessageReportedFact fixType
    WARNING: type "void bar(Test, T)" did not match /(?<raw>[^<,?!*]+)(?:<(?<args>.+)>)?(?<suffix>[?!*])?/
````

Maybe these are because I output types for methods and that pattern doesn't work for that?
I can filter out method invocations to avoid those, but there are also similar messages about wildcards on `main`.

Why do we need to fix up types? Couldn't we use a `TypeFormatter` that outputs the information in the needed format, by (optionally) modifying https://github.com/jspecify/jspecify-reference-checker/blob/ec180c5af659ac9c242672a8a0e84c63090dd195/src/main/java/com/google/jspecify/nullness/NullSpecAnnotatedTypeFactory.java#L1542 ?

When trying to compare against the output on `main`, I noticed #135.

Reply:
Yes, we see those `NullnessUnspecified` errors on main. Hence https://github.com/jspecify/jspecify/issues/361. That's one of the differences between main and samples-google-prototype, which (still) has that annotation.

You can ignore the warnings about fixing types. Those are there to remind us that the [`fixType`](https://github.com/jspecify/jspecify-reference-checker/blob/ec180c5af659ac9c242672a8a0e84c63090dd195/src/test/java/tests/ConformanceTest.java#L243) method is incomplete. I don't know whether you should emit types for entire methods. But unless you're doing extra work to emit those, I think it would be better to filter in the conformance test itself. In any case, they're just warnings.

We could emit the types in the format that the JSpecify tests expect, but I thought it would be better to put test-specific logic into the test, rather than into the analysis tool itself. We can discuss that decision.

Reply:
Ah, that makes sense. Let's discuss the type representation separately.

I think many of the new failures are caused by these compilation failures.
I do not see why these compilation issues only arise with this PR and not before on `main-eisop`.
The numbers go from 75 samples passing on `main` to 67 samples passing with this PR, which seems okay.
So I think this PR fixes the source/sink type output.